### DISCUSSION TITLE
Understanding Starvation

### DISCUSSION THREAD

**Discussion by LRagji** [jasnell/piscina#179] [Q&A]
This is a good library and provides great abstraction to underlying worker methods so thanks for the effort, recently i was monitoring my application with the properties that have been provided on the class and i am seeing a strange pattern can someone explain what does it mean .. 

`Total Time required by Async Method : 6086ms`
`Total Threads : 9`
`Utilisation: 69%`
`Waiting Time: 4208ms`
`Runtime Average: 604ms`
and this repeats .....

- Is my application starving on threads? 
- Why is there a huge wait time when runtime time is so small?
- If everything is good then why is the method taking 6 second instead of the actual runtime + some tolerance?

Config `idleTimeout:3000; concurrentTasksPerWorker:1`

**Comment by jasnell** [comment]
The total time includes waiting time, runtime, and the messaging passing overhead which depends quite a bit on the timing of the event loop. If you enable [event loop delay monitoring](https://nodejs.org/dist/latest-v17.x/docs/api/perf_hooks.html#perf_hooksmonitoreventloopdelayoptions) it might give you an idea of how much latency event loop delay is introducing.

**Reply by LRagji** [reply]
Which would explain the “total time” composition, 
What still seems to be missing is why is there a wait time when the utilisation is just 69% ?  Should it not be 100% for wait time to be more than 0?
